core/src/main/java/org/elasticsearch/action/suggest/SuggestRequestBuilder.java

```diff
@@ -45,7 +45,7 @@ public SuggestRequestBuilder(ElasticsearchClient client, SuggestAction action) {
     /**
      * Add a definition for suggestions to the request
      */
-    public <T> SuggestRequestBuilder addSuggestion(SuggestionBuilder<T> suggestion) {
+    public <T extends SuggestionBuilder> SuggestRequestBuilder addSuggestion(SuggestionBuilder<T> suggestion) {
```

Comment:
The reason the generic type had to extend SuggestionBuilder was because in the SuggestionBuilder#readFrom() method, when I read from the sub-type, I need to know it is an instance of SuggestionBuilder in order to populate the remaining SuggestionBuilder fields.  An alternative is to have each sub-type be responsible for the readFrom method and calling the super type's readFrom, but I didn't feel like that was a good alternative either.  I'm open to discussing alternative suggestions.